custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -0,0 +1,273 @@
+    treatment_support_designation,
+    patient_type_choice,
+    disease_classification,
+    dcexpulmonory,
+    dcpulmonory,
+)
+from custom.enikshay.case_utils import update_case
+
+ENIKSHAY_ID = 8
+
+
+@RegisterGenerator(NikshayRegisterPatientRepeater, 'case_json', 'JSON', is_default=True)
+class NikshayRegisterPatientPayloadGenerator(BasePayloadGenerator):
+    @property
+    def content_type(self):
+        return 'application/json'
+
+    def get_payload(self, repeat_record, episode_case):
+        """
+        mandatory_fields for Nikshay Registration API as of now
+        scode
+        dcode
+        pname
+        pgender
+        paddress
+        pmob
+        ptbyr
+        pregdate
+        cname
+        caddress
+        cmob
+        dcpulmunory
+        dotname
+        dotdesignation
+        dotmob
+        dotpType
+        dotcenter
+        regBy # Used date_of_diagnosis instead of date_of_registration
+        IP_From # Pending
+        Local_ID
+        tcode # Pending
+        page
+        poccupation
+        dotphi
+        Ptype
+        pcategory # Pending
+        Source
+        """
+        # properties skipped with example values
+        # "plandline": "N/A",
+        # "clandline": "N/A",
+        # "dotlandline": "N/A",
+        # "cvisitedby": "N/A",
+        # "cvisitedDate": "N/A",
+        # "dotcenter": "N/A",
+        # "dotmoname": "N/A",
+        # "mosign": "N/A",
+        # "mosigndonedate": "N/A",
+        # "atbtreatment": "N/A",
+        # "atbduration": ,
+        # "atbsource": "G",
+        # "atbregimen": "Manish-Regim",
+        # "atbyr": "2015",
+        # "dcpulmunory": "P",
+        # "IP_Address": "mk-ip-address",
+        # "IP_From": "127.0.0.1",
+        person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+
+        episode_case_properties = episode_case.dynamic_case_properties()
+        person_case_properties = person_case.dynamic_case_properties()
+        properties_dict = {
+            "regBy": CouchUser.get(person_case.opened_by).human_friendly_name,
+            "Local_ID": person_case.get_id,
+            "Source": ENIKSHAY_ID,
+            "dotcenter": "NA",
+            "IP_From": "127.0.0.1",
+        }
+        try:
+            properties_dict.update(_get_person_case_properties(person_case, person_case_properties))
+        except NikshayLocationNotFound as e:
+            _save_error_message(person_case.domain, person_case.case_id, e)
+        properties_dict.update(_get_episode_case_properties(episode_case_properties))
+        return json.dumps(properties_dict)
+
+    def handle_success(self, response, payload_doc, repeat_record):
+        # A success would be getting a nikshay_id for the patient
+        # without it this would actually be a failure
+        try:
+            nikshay_id = _get_nikshay_id_from_response(response)
+            update_case(
+                payload_doc.domain,
+                payload_doc.case_id,
```

Comment:
@proteusvacuum would this be the episode case? I kept this on person case since these properties `nikshay_registered` etc resonated better with it than episode.